src/entities/abTrail/ui/AbCanvas.tsx

```diff
@@ -351,13 +359,10 @@ const AbCanvas: FC<Props> = props => {
 
```

Comment:
I wonder if it's possible to just add onResult() here, before all the if-statements